src/btree/bt_sync.c

```diff
@@ -64,6 +58,20 @@ __sync_checkpoint_can_skip(WT_SESSION_IMPL *session, WT_REF *ref)
             if (multi->addr.addr == NULL)
                 return (false);
 
+     /* RTS, recovery or shutdown should not leave anything dirty behind. */
+    if (F_ISSET(session, WT_SESSION_ROLLBACK_TO_STABLE))
+        return (false);
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING | WT_CONN_CLOSING_CHECKPOINT))
+        return (false);
+
+    /*
+     * There is no snapshot transaction active. Usually, there is one in ordinary application
+     * checkpoints but not all internal cases. Furthermore, this guarantees the metadata file is
```

Comment:
@kommiharibabu, if this one is about not skipping the metadata file, is it ok to have this check at the end? Isn't this code executed a lot of the metadata file? Why are we not checking if we are dealing with the metadata file in a separate check?

Reply:
Yes, that flow is possible only for the metadata. I'm not sure why we didn't check directly for the metadata. Checking at the end is fine.